syntax = "proto3";

package jobreviews;

// Represents a company and its reviews.
message CompanyReview {
  string firm = 1;
  int32 overall_rating = 2;
  float work_life_balance = 3;
  float culture_values = 4;
  float diversity_inclusion = 5;
  float career_opp = 6; // Defaults to 0 if not provided.
}

// -----------------------------
// Best Company Messages
// -----------------------------
message BestCompaniesRequest {
}

message BestCompaniesResponse {
  repeated CompanyReview companyReview = 1;
}

// -----------------------------
// Service Definition
// -----------------------------

// Service handling job posting functionalities.
service JobReviewService {
  rpc GetBestCompanies(BestCompaniesRequest) returns (BestCompaniesResponse);
}